### PR TITLE
Feature/task condition

### PR DESCRIPTION
**Pull request by vkirichenko**

### CODE REVIEW

File: src/main/java/com/icthh/xm/tmf/ms/activation/service/SagaServiceImpl.java [hunk 1]
```diff
@@ -230,6 +231,26 @@ private boolean isTransactionInCorrectState(SagaEvent sagaEvent, Context context
...
+        try {
+            boolean taskExecutionAllowed = taskExecutor.checkWaitCondition(taskSpec, sagaEvent, transaction);
+            if (!taskExecutionAllowed) {
+                log.info("Task will not executed. Wait condition not happened yet. Transaction id {}.", txId);
+                retryService.retryForTaskWaitCondition(sagaEvent, context.getTaskSpec());
+                return false;
+            }
+        } catch (Throwable t) {
+            log.error("Task will not executed. Error during condition check. Transaction id {}.", txId);
+            retryService.retryForTaskWaitCondition(sagaEvent, context.getTaskSpec());
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/tmf/ms/activation/service/SagaServiceImpl.java:248]: better in this case it's retry with state ON_RETRY

File: src/main/java/com/icthh/xm/tmf/ms/activation/service/RetryService.java [hunk 2]
```diff
@@ -44,6 +45,7 @@ public void postConstruct() {
     public void rescheduleAllEvents() {
         sagaEventRepository.findByStatus(ON_RETRY).forEach(this::doResend);
         sagaEventRepository.findByStatus(WAIT_DEPENDS_TASK).forEach(this::doResend);
+        sagaEventRepository.findByStatus(TASK_WAIT_CONDITION).forEach(this::doResend);
```
Comment (amedvedchuk) [inline src/main/java/com/icthh/xm/tmf/ms/activation/service/RetryService.java:48]: suggest rename `TASK_WAIT_CONDITION` --> `WAIT_TASK_CONDITION`

Comment (amedvedchuk) [reply src/main/java/com/icthh/xm/tmf/ms/activation/service/RetryService.java:48]: or maybe even to `WAIT_CONDITION_TASK`?

File: src/main/java/com/icthh/xm/tmf/ms/activation/service/SagaTaskExecutor.java [hunk 3]
```diff
@@ -25,6 +25,12 @@
 
     @LogicExtensionPoint(value = "OnFinish", resolver = TransactionTypeKeyResolver.class)
     public void onFinish(SagaTransaction sagaTransaction) {
-        log.warn("Script for finish not found. Transaction {}.", sagaTransaction);
+        log.info("Script for finish not found. Transaction {}.", sagaTransaction);
+    }
+
+    @LogicExtensionPoint(value = "TaskCheckWaitCondition", resolver = TaskTypeKeyResolver.class)
```
Comment (amedvedchuk) [inline src/main/java/com/icthh/xm/tmf/ms/activation/service/SagaTaskExecutor.java:31]: propose rename `TaskCheckWaitCondition` to `OnCheckWaitCondition`

File: src/main/java/com/icthh/xm/tmf/ms/activation/service/SagaServiceImpl.java (hunk 1 above)
Comment (vkirichenko) [reply src/main/java/com/icthh/xm/tmf/ms/activation/service/SagaServiceImpl.java]: done

File: src/main/java/com/icthh/xm/tmf/ms/activation/service/RetryService.java (hunk 2 above)
Comment (vkirichenko) [reply src/main/java/com/icthh/xm/tmf/ms/activation/service/RetryService.java]: done

File: src/main/java/com/icthh/xm/tmf/ms/activation/service/SagaTaskExecutor.java (hunk 3 above)
Comment (vkirichenko) [reply src/main/java/com/icthh/xm/tmf/ms/activation/service/SagaTaskExecutor.java]: done

**Pull request merged**